hw4.Rmd

```diff
@@ -30,43 +37,77 @@ Before doing anything else, just play with the data to get a sense of it.
 (@) The main outcome of interest is *decision* which records whether the judge granted an appeal or not.  In what proprotion of cases did the judge grant an appeal.
 
 ```{r}
-# [your code here]
+data %>%
+  summarise(prop_granted = mean(decision == "yes"))
 ```
 
 (@) There are 12 different judges in the data. A key question is whether different judges have different rates of granting an appeal.  Make a plot showing the number of cases per judge.
 
 ```{r}
-# [your code here]
+
+ggplot(data, aes(x = judge, fill = judge)) + 
+  stat_bin(type = "identity", width = 0.75) + 
+  labs(title = "The Number of Appeals per Judge", x = "Judges", y = "Appeals") +
+  coord_flip() 
+  
 ```
 
 (@) Now plot the rate at which the judges grant appeals. 
 
 ```{r}
-# [your code here]
+data <- data %>%
+  group_by(judge) %>%
+  mutate(appeal_rate = sum(decision == "yes")/length(decision))
+
+ggplot(data, aes(x = judge, y = appeal_rate, color = judge)) + 
+  geom_point(size = 5) +
+  labs(title = "Appeal Rates per Judge", x = "Judges", y = "Rate of Appeal") +
+  coord_flip()
+  
 ```
 
-(@) Now let's try this with logistic regression because we are going to move to more complicated models.  Fit a logistic regression predicting whetheran appeal will be granted using judge as a categorical predictor. Use Iacobucci as the reference judge to facilitate the comparisons that follow. For more on how to control the reference level of a factor, check out this post on StackOverflow: http://stackoverflow.com/questions/3872070/how-to-force-r-to-use-a-specified-factor-level-as-reference-in-a-regression
+(@) Now let's try this with logistic regression because we are going to move to more complicated models.  Fit a logistic regression predicting whether an appeal will be granted using judge as a categorical predictor. Use Iacobucci as the reference judge to facilitate the comparisons that follow. For more on how to control the reference level of a factor, check out this post on StackOverflow: http://stackoverflow.com/questions/3872070/how-to-force-r-to-use-a-specified-factor-level-as-reference-in-a-regression
 
 ```{r}
-# [your code here]
+levels(data$judge)
+data <- within(data, judge <- relevel(judge, ref = "Iacobucci"))
+levels(data$judge)
+
+model1 <- glm(decision ~ judge, data = data, family = "binomial")
+model1 <- tidy(model1)
 ```
 
 (@) Using the results of the model calculate the predicted probability that Judge Marceau will grant an appeal.
 
 ```{r}
-# [your code here]
+# Predicted probability for Judge Marceau
+
+(exp(model1$estimate[1] + model1$estimate[7])) / (1 + exp(model1$estimate[1] + model1$estimate[7]))
```

Comment:
Try not to call estimate with position numbers, but instead by a name which is more robust and less prone to errors. See solutions for an example.